src/main/java/org/broadinstitute/hellbender/engine/spark/SparkContextFactory.java

```diff
@@ -153,4 +156,36 @@ private static JavaSparkContext createTestSparkContext(Map<String, String> overr
         final SparkConf sparkConf = setupSparkConf("TestContext", DEFAULT_SPARK_MASTER, DEFAULT_TEST_PROPERTIES, overridingProperties);
         return new JavaSparkContext(sparkConf);
     }
+
+    /**
+     * Determine the number of cores Spark master should use. Applicable to Spark test only.
+     *   Read the specification from the environmental variable GATK_TEST_SPARK_CORES
+     *      If the value is a valid positive integer, use the value
+     *      If the enviromental variable is not set,  use all available cores as in "local[*]"
+     *      If the value is invalid (strings, empty, etc), throw an UserException
+     *      If the value is a negative interger or zero, throw an UserException
+     */
+    private static String determineDefaultSparkMaster() {
+	final String defaultSparkMasterString = "local[*]";
+	String sparkMasterString;
+
+	String sparkSpecFromEnvironment = System.getenv( sparkSpecEnvName );
+	if ( null == sparkSpecFromEnvironment ) {
+	    sparkMasterString = defaultSparkMasterString;
+	} else {
+	    int numSparkCoresFromEnv = 0;
+	    try {
+		numSparkCoresFromEnv = Integer.parseInt( sparkSpecFromEnvironment );
+	    } catch ( NumberFormatException e ) {
+		throw new UserException("Illegal Spark master specification found in " + sparkSpecEnvName + ". Positive integers only");
```

Comment:
"Illegal Spark master specification" -> "Illegal number of cores" (since the environment variables contains only a number, not the full Spark master String)